BlueprintUI/Sources/Layout/DecorateBackground.swift

```diff
@@ -0,0 +1,185 @@
+//
+//  DecorateBackground.swift
+//  BlueprintUI
+//
+//  Created by Kyle Van Essen on 11/4/20.
+//
+
+import UIKit
+
+
+///
+/// Places a background behind the given `wrapped` element,
+/// and overflows it by the amount specified in `overflow`.
+///
+/// The size of the element is determined only by the `wrapped` element –
+/// that is, the `background` element will overflow the bounds of the element, but not
+/// affect its layout or size. This is similar to how UIKit handles shadows: They live outside of the layout
+/// rect of the view:
+///
+/// The arrows represent the measured size of the view for layout purposes.
+/// ```
+/// ┌───────────────────┐
+/// │    Background     │
+/// │ ┏━━━━━━━━━━━━━━━┓ │ ▲
+/// │ ┃               ┃ │ │
+/// │ ┃    Wrapped    ┃ │ │
+/// │ ┃               ┃ │ │
+/// │ ┗━━━━━━━━━━━━━━━┛ │ ▼
+/// └───────────────────┘
+///   ◀───────────────▶
+/// ```
+///
+/// This is useful to render on-touch or selected states of elements, where you
+/// want to provide a padded background that does otherwise affect the layout of the element,
+/// which is likely controlled by its parent element.
+///
+public struct DecorateBackground : Element {
+    
+    /// The element which provides the sizing and measurement.
+    public var wrapped : Element
+    
+    /// The element which is used to render the background.
+    /// It is stretched to fit the `wrapped` content, plus the `overflow` padding.
+    ///
+    /// If you have a 100w x 50h element, and an overflow of (10, 10, 10, 10),
+    /// the measured sized will be 100w x 50h, and the background will be
+    /// sized to be 120w x 70h.
+    /// ```
+    /// ┌───────────────────┐
+    /// │ ┏━━━━━━━━━━━━━━━┓ │ ▲
+    /// │ ┃               ┃ │ │
+    /// │ ┃    Wrapped    ┃ │ │
+    /// │ ┃               ┃ │ │
+    /// │ ┗━━━━━━━━━━━━━━━┛ │ ▼
+    /// └───────────────────┘
+    ///   ◀───────────────▶
+    /// ```
+    public var background : Element
+    
+    /// How much the background should overflow the measured bounds of the
+    /// element. Positive values overflow outside of the bounds, and negative
+    /// values underflow to inside the bounds.
+    public var overflow : UIEdgeInsets
+    
+    /// Creates a new instance with the provided overflow, background, and wrapped element.
+    public init(
+        overflow: UIEdgeInsets,
+        background: () -> Element,
```

Comment:
`@autoclosure` maybe? Curious why you're switching up the API from what most of the elements do, especially since you're just immediately calling the func?